### PR TITLE
Add editorconfig

### PR DESCRIPTION
**Pull request by NeilGirdhar**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+max_line_length = 88
+indent_size = 4
+
+[*.toml]
+max_line_length = 150
+indent_size = 4
+
+[*.yml]
+max_line_length = 120
+indent_size = 2
```
Comment (patrick-kidger) [inline .editorconfig:19]: I'd cut these two. Right now we rarely use either kind of file and both are unformatted.

**Review by patrick-kidger — COMMENTED** [review]
Do you feel that this adds much over the existing autoformatting (via pre-commit + ruff)? It feels more natural to me to have just a single place that handle this.

**Comment by NeilGirdhar** [comment]
editorconfig automatically configures editors like vim, emacs, etc. so that they know how to save files, how long the lines should be (for examples, vim's automatic line width formatter), where to show the line length helper.

File: .editorconfig (hunk 1 above)
Comment (NeilGirdhar) [reply .editorconfig]: This is for editors, not formatters.  The indent size tells the editor how many spaces to add when <tab> is pressed.    I got rid of the line length then.

**Comment by NeilGirdhar** [comment]
You can see a list of editors that support this file here: https://editorconfig.org/

**Comment by patrick-kidger** [comment]
Indeed, I'm familiar with `.editorconfig`.

My point is that if we care about such things then IMO it'd be better to have a single source of truth on such matters -- an autoformatter -- and allow anything at all whilst editing.

Anyway, not a strong opinion. `*.py` files are the only ones we attempt to format so trim this down to just that and I'd be happy to merge this.

**Comment by NeilGirdhar** [comment]
> My point is that if we care about such things then IMO it'd be better to have a single source of truth on such matters -- an autoformatter -- and allow anything at all whilst editing.

Yes, I see your point.  I guess I forgot that you've got the autoformatter enabled.

Anyway, updated as requested.

**Comment by patrick-kidger** [comment]
Alright, merged! :) Thank you for the contribution!

**Comment by NeilGirdhar** [comment]
Thanks for being flexible about this.  Maybe we should propose to Ruff that editorconfig be the defaults to satisfy your "one source of truth" desire?

**Comment by patrick-kidger** [comment]
Seems not planned: https://github.com/astral-sh/ruff/issues/1530

**Comment by NeilGirdhar** [comment]
Ah, nice find!

**Pull request merged**